src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationRefresherProvider.cs

```diff
@@ -26,20 +39,23 @@ public AzureAppConfigurationRefresherProvider(IConfiguration configuration, ILog
                         // Use _loggerFactory only if LoggerFactory hasn't been set in AzureAppConfigurationOptions
                         if (refresher.LoggerFactory == null)
                         {
-                            refresher.LoggerFactory = _loggerFactory;
+                            refresher.LoggerFactory = loggerFactory;
                         }
 
                         refreshers.Add(refresher);
                     }
-                }
-            }
+                    else if (provider is ChainedConfigurationProvider chainedProvider)
+                    {
+                        PropertyInfo propertyInfo = typeof(ChainedConfigurationProvider).GetProperty("Configuration", BindingFlags.Public | BindingFlags.Instance);
```

Comment:
Why reflection?

Reply:
I wasn't really sure how to approach this, I'm only aware of reflection or using predefined symbols like `#if NET7_0` to identify the version. There just has to be a way to identify whether this version has the `Configuration` property.

Reply:
The two approaches have different behaviors at runtime. Do we need this to happen on runtimes other than .net 7?

Reply:
If we use net7_0 macro, we wouldn't be able to extend this feature to future versions of .NET, right?

Reply:
There are all kinds of problems with version detection, so feature detection (using reflection on public API) is normally preferred. Please make sure the reflection won't occur repeatedly.

Reply:
There is a net7_0_or_greater symbol for the version issue, but since it sounds like that's not preferred I'll go with reflection. I don't think it's called more than needed right now for this issue.

Reply:
PropertyInfo can be a private field so that we only do this feature detection once.

Reply:
Oh that makes sense actually